tests/unittests/tests-dns_msg/tests-dns_msg.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2024 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include "net/af.h"
+#include "net/ipv6.h"
+#include "ztimer.h"
+
+#include "net/dns/msg.h"
+
+#include "tests-dns_msg.h"
+
+static void test_dns_msg_valid_google(void)
+{
+    const uint8_t dns_msg[] = {
+        0x00, 0x00, 0x81, 0x80, 0x00, 0x01, 0x00, 0x01,
+        0x00, 0x00, 0x00, 0x00, 0x06, 0x67, 0x6f, 0x6f,
+        0x67, 0x6c, 0x65, 0x03, 0x63, 0x6f, 0x6d, 0x00,
+        0x00, 0x1c, 0x00, 0x01, 0xc0, 0x0c, 0x00, 0x1c,
+        0x00, 0x01, 0x00, 0x00, 0x01, 0x2c, 0x00, 0x10,
+        0x2a, 0x00, 0x14, 0x50, 0x40, 0x05, 0x08, 0x0b,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x20, 0x0e,
+    };
+    const uint8_t addr[] = {
+        0x2a, 0x00, 0x14, 0x50, 0x40, 0x05, 0x08, 0x0b,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x20, 0x0e,
```

Comment:
I don't know how to generate synthetic DNS response messages. Also having real-world test data in there is a good way to verify our implementation against what's out there - I don't think there is copyright on DNS response messages.

Reply:
…but why?

Reply:
Arghs, just realized that this did not compress the name... use

```python
str([f"0x{b:02x}" for b in bytes(dns.compress())]).replace("'", "").replace("[", "{").replace("]", "}")
```

in the last line

Reply:
This seems like a lot of busywork for no real gain.
Why not use real world DNS messages to test against?

Reply:
> …but why?

Because it is unnecessary meta-data.

Reply:
It was unnecessary meta-data (two CNAME records) that made the parsing fail in the first place!

Reply:
How? Your OP doesn't mention them. And even then, afair your original bug analysis it was the length of the CNAME record that caused the issue, not the value itself.

Reply:
> your original bug analysis it was the length of the CNAME record that caused the issue

The issue was that the buffer pointer was not advanced when a record was skipped, so `bufpos` still pointed to the middle of the previous record.

Reply:
And how does the content of the CNAME fits into this? _confused_

Reply:
Also: If it is so hard, even for you as a test writer, to explain the test case, maybe it would indeed be better, if the DNS message would be generated in-code by, e.g. `scapy` (which AFAIK is installed on the pifleet), so that future test readers can discern what these test vectors actually test. At least add some comments on what the DNS message contains.